src/Widgets/AbstractAppContainer.vala

```diff
@@ -80,6 +82,19 @@ namespace AppCenter {
             }
         }
 
+        private string? suggested_amount_ = null;
+        public string suggested_amount {
+            get {
+                if (this.package == null || this.package.component == null || updates_view) {
+                    return DEFAULT_PRICE_DOLLARS;
+                } else if (suggested_amount_ == null) {
+                    suggested_amount_ = this.package.component.get_custom_value ("x-appcenter-suggested-price");
+                }
+
+                return suggested_amount_ != null ? suggested_amount_ : DEFAULT_PRICE_DOLLARS;
+            }
+        }
+
```

Comment:
you could pass it as a construction parameter to the HumbleButton then instead of forcing the amount at the end there :)